src/appleseed/renderer/kernel/shading/shadingpoint.cpp

```diff
@@ -967,13 +967,60 @@ void ShadingPoint::compute_alpha() const
 
       case PrimitiveCurve1:
       case PrimitiveCurve3:
-        // todo: interpolate per vertex alpha for curves here.
+        {
+            assert(is_curve_primitive());
+
+            const GScalar v = m_bary[1];
+
+            if (const CurveObject* curves = static_cast<const CurveObject*>(&get_object()))
```

Comment:
Is this `if` really necessary? Under which conditions can it fail?

Reply:
I guess it isn't necessary. Again overkill code. 👎